pkg/result/processors/exclusion_rules.go

```diff
@@ -0,0 +1,190 @@
+package processors
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/fsutils"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+var _ Processor = (*ExclusionRules)(nil)
+
+type excludeRule struct {
+	baseRule
+}
+
+func (e excludeRule) String() string {
+	var msg []string
+
+	if e.text != nil && e.text.String() != "" {
+		msg = append(msg, fmt.Sprintf("Text: %q", e.text))
+	}
+
+	if e.source != nil && e.source.String() != "" {
+		msg = append(msg, fmt.Sprintf("Source: %q", e.source))
+	}
+
+	if e.path != nil && e.path.String() != "" {
+		msg = append(msg, fmt.Sprintf("Path: %q", e.path))
+	}
+
+	if e.pathExcept != nil && e.pathExcept.String() != "" {
+		msg = append(msg, fmt.Sprintf("Path Except: %q", e.pathExcept))
+	}
+
+	if len(e.linters) > 0 {
+		msg = append(msg, fmt.Sprintf("Linters: %q", strings.Join(e.linters, ", ")))
+	}
+
+	return strings.Join(msg, ", ")
+}
+
+type ExclusionRules struct {
+	log   logutils.Log
+	files *fsutils.Files
+
+	warnUnused     bool
+	skippedCounter map[string]int
+
+	rules []excludeRule
+}
+
+func NewExclusionRules(log logutils.Log, files *fsutils.Files, cfg *config.LinterExclusions, oldCfg *config.Issues) *ExclusionRules {
+	p := &ExclusionRules{
+		log:            log,
+		files:          files,
+		warnUnused:     cfg.WarnUnused,
+		skippedCounter: map[string]int{},
+	}
+
+	// TODO(ldez) remove prefix in v2: the matching must be case sensitive, users can add `(?i)` inside the patterns if needed.
+	prefix := caseInsensitivePrefix
+	if oldCfg.ExcludeCaseSensitive {
+		prefix = ""
+	}
+
+	excludeRules := slices.Clone(cfg.Rules)
+	excludeRules = append(excludeRules, filterInclude(getDefaultLintersExclusions(cfg.Default), oldCfg.IncludeDefaultExcludes)...)
+
+	p.rules = createRules(excludeRules, prefix)
+
+	// TODO(ldez): should be remove in v2.
```

Comment:
Nit

```suggestion
	// TODO(ldez): should be removed in v2.
```

You wrote it randomly at a few place